### ISSUE TITLE
Can the documentation be optimized?

### ISSUE THREAD

**Issue by medz** [nhost/nhost#1817]
The nhost documentation is very unfriendly to back-end personnel, and the nhost documentation can be described as "simple".
The ubiquitous frontends on how to use it provide hardly any helpful documentation for serverless function writing etc.

For example:
- How to depend on other 3rd party packages in serverless functions?
- How to call nhost endpoint service in serverless function.
...There are a lot more.

I'm using nhost cloud for my project. This makes it very painful and annoying and irritating.
Because I can't retrieve any information from Google or other search engines that is helpful for writing serverless functions with nhost.

**Comment by dbarrosop** [comment]
What information are you missing here?

https://docs.nhost.io/serverless-functions

> How to depend on other 3rd party packages in serverless functions?

you just install them with yarn/npm/pnpn like in the example (but without the -D):

https://docs.nhost.io/serverless-functions#create-a-serverless-function

Just make sure you commit your packages file, there is nothing special here, just standard js/ts development.

> How to call nhost endpoint service in serverless function

https://docs.nhost.io/serverless-functions#routing

> There are a lot more

for instance?

> I'm using nhost cloud for my project. This makes it very painful and annoying and irritating.

Sorry to hear that, if you have suggestions on how to improve feel free to let us know. We are also open to PRs improving on the docs.

**Comment by medz** [comment]
@dbarrosop The more function of the serverless function is to expand the actual business of nhost. I have the following problem and can't find the documentation:

1. Customized SSO authentication login extension
2. The ability to manage nhost data and functions in serverless functions (Manager SDK?)
3. Independent dependencies in serverless functions

I wrote my first nhost serverless function which depends on `@nhost/nhost-js` and `bcrypt` added to `package.json` dependencies. Unfortunately, the serverless function build fails. It doesn't seem to work like a normal node program.

Also, for the admin graphql endpoint, the need to add `x-hasura-admin-secret` to the header is not mentioned.

I have used products such as firebase, cloudbase, appwrite, etc., and they have written better descriptions of the most commonly used scenarios of no-service (except appwrite), including how to call GraphQL endpoints in no-service. It will also write some documentation that broadens the SSO or social login provider, rather than just writing how front-end developers use their documentation content.

The nhost documentation could be better, although the current documentation is friendly to front-end people who just use it. But for the backend or operation and maintenance personnel, we have to read a very large source code. Or even come up with impractical but effective ways to use it as a front-end service.
